test/e2e/specs/site-editor/template-part.spec.js

```diff
@@ -356,4 +356,50 @@ test.describe( 'Template Part', () => {
 			page.getByRole( 'combobox', { name: 'Import widget area' } )
 		).not.toBeVisible();
 	} );
+
+	test( 'Keeps focus in place on undo in template parts', async ( {
+		admin,
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'emptytheme//header',
+			postType: 'wp_template_part',
+		} );
+		await editor.canvas.click( 'body' );
+
+		// Select the site title block.
+		const siteTitle = editor.canvas.getByRole( 'document', {
+			name: 'Site title',
+		} );
+		await editor.selectBlocks( siteTitle );
+
+		await pageUtils.pressKeys( 'access+z' );
+
+		// Insert a group block
+		await editor.insertBlock( {
+			name: 'core/group',
+			innerBlocks: [ { name: 'core/site-title' } ],
+		} );
+
+		// Select the site title block.
+		const siteTitleInGroup = editor.canvas.getByRole( 'document', {
+			name: 'Site title',
+		} );
+		await editor.selectBlocks( siteTitleInGroup );
+
+		// Change heading level.
+		await editor.clickBlockToolbarButton( 'Change heading level' );
+		const Heading3Button = page.locator(
+			'role=menuitemradio[name="Heading 3"i]'
+		);
+		await Heading3Button.click();
+
+		await pageUtils.pressKeys( 'primary+z' );
+
+		await expect(
+			page.locator( 'role=button[name="Change heading level"i]' )
+		).toBeFocused();
```

Comment:
```suggestion
		await expect(
			page.getByRole( 'button', {
				name: 'Change heading level',
			} )
		).toBeFocused();
```

As I can't seem to run locally this will need verifying...

Reply:
what is the difference?

Reply:
[The docs](https://testing-library.com/docs/) recommend prefering the built in helper queries over `locator` where possible.

I was also advised on this by @kevin940726 who may have additional information for you.

Reply:
This is an awkward one 😅.

> what is the difference?

They really are the same here. `getByRole` and its variants are the new API and before that we're using the `role=` selector. Since Playwright decided to remove the doc for `role=` selector and recommended `getByRole` instead, we should also follow that rule for readability's sake.

I don't think they will deprecate the text-based `role=` selector though since it's being used under the hood. But we might add some lint rules to warn against using `locator` at all, but that should only be enabled when we migrate all `locator` to `getByRole`.

Hence, in conclusion, both are fine, but `getByRole` is preferred if you're writing new tests :)!

Reply:
What he said 👍